gtdynamics/kinematics/KinematicsTrajectory.cpp

```diff
@@ -0,0 +1,101 @@
+/* ----------------------------------------------------------------------------
+ * GTDynamics Copyright 2020, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file  KinematicsInterval.cpp
+ * @brief Kinematics for an trajectory with fixed contacts.
+ * @author: Frank Dellaert
+ */
+
+#include <gtdynamics/kinematics/Kinematics.h>
+#include <gtdynamics/utils/Slice.h>
+#include <gtdynamics/utils/Trajectory.h>
+#include <gtsam/nonlinear/GaussNewtonOptimizer.h>
+#include <gtsam/nonlinear/LevenbergMarquardtOptimizer.h>
+
+namespace gtdynamics {
+
+using gtsam::NonlinearFactorGraph;
+using gtsam::Values;
+using std::string;
+using std::vector;
+
+template <>
+NonlinearFactorGraph Kinematics::graph<Trajectory>(const Trajectory& trajectory,
+                                                   const Robot& robot) const {
+  NonlinearFactorGraph graph;
+  for (auto&& phase : trajectory.phases()) {
+    graph.add(this->graph(phase, robot));
+  }
+  return graph;
+}
+
+template <>
+NonlinearFactorGraph Kinematics::pointGoalObjectives<Trajectory>(
+    const Trajectory& trajectory, const ContactGoals& contact_goals) const {
+  NonlinearFactorGraph graph;
+  for (size_t k = trajectory.getStartTimeStep(0); k <= trajectory.getEndTimeStep(trajectory.phases().size()-1); k++) {
+    graph.add(pointGoalObjectives(Slice(k), contact_goals));
```

Comment:
Hmmm, should this not call a function in KinematicsPhase.cpp? Surprised to see a direct useage of Slice(k) here, which is my red flag

Reply:
added KinematicsPhase.cpp